widget/table.go

```diff
@@ -487,7 +487,8 @@ func (t *Table) columnAt(pos fyne.Position) int {
 		pos.X += t.content.Offset.X
 		offX += t.stuckXOff
 	}
-	for x := offX; i < end; x += visibleColWidths[i-1] + theme.Padding() {
+	padding := theme.Padding()
```

Comment:
I admit I had assumed the compiler would perform this optimisation!

Reply:
It actually can not and never will be able to. As it is behind a lock and shared across multiple goroutine, it can not know when it is acceptable to reuse the previous result of the call and has to assume the value changed between two calls. 

In general, if the compiler can not prove that the value it is accessing can not be changed by anyone else, it will redo a fetch.